### PR TITLE
feat: warn for missing response examples

### PR DESCRIPTION
**Pull request by barrett-schonefeld**
Purpose:
- Establish working pattern for writing custom functions such that, when users extend the `ibm:oas` ruleset, our custom functions are still findable by Spectral.
- Tooling to generate documentation relies on response examples provided either in the schema or "next to" the schema at response level. Warn about responses that do not provide examples in these locations.

Changes:
- Create a custom Spectral rule to warn about missing response examples
- Create a custom Spectral function to check if a response example provided in the locations supported by the code that generates documentation.

Tests:
- Add tests to ensure no warning given when a response example is provided.
- Add a test to ensure a warning is given when no response example provided.

Docs:
- Document the response-example-provided rule in `docs/spectral-rules.md`

**Possible changes:**
- Include more than `application/json` responses. To do this, we would change the `given` field of the rule from `$.paths[*][*].responses[*].content.application/json` to `$.paths[*][*].responses[*].content[*]`.
- May want to condense the error message. I had writer's block, so I went with a verbose description initially.

### CODE REVIEW

File: src/spectral/rulesets/ibm-oas/response-example-provided.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+module.exports = function(response) {
+  if (!responseLevelExamples(response) && !schemaLevelExample(response)) {
+    return [
+      {
+        message:
+          'To generate good documenation, should provide response example(s) inside of or at the the same level as the schema'
```
Comment (dpopp07) [inline src/spectral/rulesets/ibm-oas/response-example-provided.js:6]: This is a little long, but it's probably fine. I will make some minor suggestions:

```suggestion
          'To generate good documentation, provide response example(s) inside of, or at the the same level as, the schema'
```

**Review by dpopp07 — APPROVED** [review]
Looks good to me!

File: src/spectral/rulesets/ibm-oas/response-example-provided.js (hunk 1 above)
Comment (mkistler) [reply src/spectral/rulesets/ibm-oas/response-example-provided.js:6]: How about:
```suggestion
          'Response bodies should include an example response'
```

I don't think we need "To generate good documentation" -- that is the goal of many of our rules (like missing descriptions) but we don't explicitly state that in the message.

And I don't think the message needs to say where the example should be specified -- we say that in the documentation of the rule.

File: test/spectral/tests/spectral-config.test.js [hunk 2]
```diff
@@ -1,37 +1,23 @@
-const path = require('path');
```
Comment (mkistler) [inline test/spectral/tests/spectral-config.test.js:1]: It looks like you snuck some unrelated changes into this PR.  I don't have a problem with that but I think you should alert reviewers to this by either mentioning it in the PR description or simply commenting on your own PR to call this out.

**Review by mkistler — CHANGES_REQUESTED** [review]
This looks good but there's one more detail that you will need to handle.  In OpenAPI 3.0 they introduced `examples` on parameters, request bodies, and response bodies.  `examples` is a slightly more elaborate way to provide an example -- you can include not just the example value but also a description and summary.  You can also have multiple named examples.

So in addition to checking for `example` you should also check for `examples` and only issue the message if neither is present.

File: test/spectral/tests/spectral-config.test.js (hunk 2 above)
Comment (barrett-schonefeld) [reply test/spectral/tests/spectral-config.test.js:1]: Yes, I will do a better job of pointing out and explaining these other changes. These tests were timing out unpredictably (maybe 1/10 or 1/20 times). The mocking seemed to cause the problems, so I removed the mocking to simplify the tests and fix the timeouts.

**Comment by mkistler** [comment]
Thanks Barrett, I missed that.  Looks like you have it covered.  Nice work!

**Review by mkistler — APPROVED** [review]
Looks good! 👍
